src/hardware/serial_terminal.rs

```diff
@@ -25,28 +85,128 @@ impl Write for OutputBuffer {
     }
 }
 
+impl core::fmt::Write for Context {
+    /// Write data to the serial terminal.
+    ///
+    /// # Note
+    /// The terminal uses an internal buffer. Overflows of the output buffer are silently ignored.
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
+        self.output.write_str(s)
+    }
+}
+
+fn handle_list(
+    _menu: &menu::Menu<Context>,
+    _item: &menu::Item<Context>,
+    _args: &[&str],
+    context: &mut Context,
+) {
+    writeln!(context, "Available properties:").unwrap();
+    for path in Settings::iter_paths::<heapless::String<32>>("/") {
+        let path = path.unwrap();
+        writeln!(context, "* {path}").unwrap();
+    }
+}
+
+fn handle_reset(
+    _menu: &menu::Menu<Context>,
+    _item: &menu::Item<Context>,
+    _args: &[&str],
+    _context: &mut Context,
+) {
+    cortex_m::peripheral::SCB::sys_reset();
+}
+
+fn handle_property_read(
+    _menu: &menu::Menu<Context>,
+    item: &menu::Item<Context>,
+    args: &[&str],
+    context: &mut Context,
+) {
+    let mut buf = [0u8; 256];
+    if let Some(path) = menu::argument_finder(item, args, "property").unwrap() {
+        let len = match context.flash.settings.get_json(path, &mut buf) {
+            Err(e) => {
+                writeln!(context, "Failed to read {path}: {e}").unwrap();
+                return;
+            }
+            Ok(len) => len,
+        };
+
+        let stringified = core::str::from_utf8(&buf[..len]).unwrap();
+        writeln!(context, "{path}: {stringified}").unwrap();
+    };
+
+    for path in Settings::iter_paths::<heapless::String<32>>("/") {
```

Comment:
Ah. You had this. Let's remove it from `read` and just keep the dump in `list`.

Reply:
Fixed in https://github.com/quartiq/stabilizer/pull/805/commits/6f171975624442b05a68dee490f8802ecc7af40a